### PR TITLE
[1.27] releng: Update release branch jobs for 1.27

### PR DESCRIPTION
**Pull request by ramrodo**
Release cut issue: https://github.com/kubernetes/sig-release/issues/2199

/sig release
/area release-eng

cc: @cici37 @gracenng @kubernetes/release-engineering

### CODE REVIEW

**Comment by cici37** [comment]
I am not sure if we still wanna keep 1.23 since it should be out of support now?
We kept 1.22 previously because the life end time for 1.22 previously was an exception.
I am also ok with cleaning up 1.23 in a followup PR.
/cc @xmudrii on this
/lgtm
/approve

File: prow/cmd/deck/static/pr/pr.ts [hunk 1]
```diff
@@ -867,10 +867,10 @@ function createQueriesTable(prLabels: { Label: Label }[], query: ProcessedQuery)
...
 
   const body = document.createElement("tbody");
-    const row = document.createElement("tr");
-    row.appendChild(createMergeLabelCell(query.labels, true));
-    row.appendChild(createMergeLabelCell(query.missingLabels));
-    body.appendChild(row);
+  const row = document.createElement("tr");
+  row.appendChild(createMergeLabelCell(query.labels, true));
+  row.appendChild(createMergeLabelCell(query.missingLabels));
+  body.appendChild(row);
```
Comment (cici37) [inline prow/cmd/deck/static/pr/pr.ts:873]: Why those changes shown up?

File: prow/plugins/plugin-config-documented.yaml [hunk 2]
```diff
@@ -338,6 +338,13 @@ config_updater:
             key: ' '
             # Name of ConfigMap
             name: ' '
+            # PartitionedNames is a slice of names of ConfigMaps that the keys should be balanced across.
+            # This is useful when no explicit key is given and file names/paths are used as keys instead.
+            # This is used to work around the 1MB ConfigMap size limit by spreading the keys across multiple
+            # separate ConfigMaps.
+            # PartitionedNames is mutually exclusive with the "Name" field.
+            partitioned_names:
+                - ""
```
Comment (cici37) [inline prow/plugins/plugin-config-documented.yaml:347]: Is this coming from make verify?

**Comment by cici37** [comment]
Not quite sure about the changes in [prow/cmd/deck](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/deck) and [prow/plugins](https://github.com/kubernetes/test-infra/blob/master/prow/plugins). Will differ this to domain expert to review

**Comment by cici37** [comment]
FYI: I removed 1.23 from kubekins-e2e variants.yaml in https://github.com/kubernetes/test-infra/pull/29125
I assume we could remove 1.23 here as well since it has been out of support.

File: prow/cmd/deck/static/pr/pr.ts (hunk 1 above)
Comment (ramrodo) [reply prow/cmd/deck/static/pr/pr.ts:873]: These changes were added at executed:

```bash
make -C releng prepare-release-branch
```

following the [release-branch-jobs guide](https://github.com/kubernetes/test-infra/blob/master/releng/README.md#release-branch-jobs) .

File: prow/plugins/plugin-config-documented.yaml (hunk 2 above)
Comment (ramrodo) [reply prow/plugins/plugin-config-documented.yaml:347]: Not. This came from this command:

```bash
make -C releng prepare-release-branch
```

**Comment by ramrodo** [comment]
> Not quite sure about the changes in [prow/cmd/deck](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/deck) and [prow/plugins](https://github.com/kubernetes/test-infra/blob/master/prow/plugins). Will differ this to domain expert to review

@cici37 unsure if we should keep or remove these auto-generated files or code changes.

**Comment by cici37** [comment]
From a glance, an unexpected 1.28 job was also been added which failed the CI.
I would suggest to try:
1. revert the changes in [prow/cmd/deck](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/deck) and [prow/plugins](https://github.com/kubernetes/test-infra/blob/master/prow/plugins) (it looks not relevant with things we are trying to do here. Make sure `make verify` not complain.)
2. Remove [release-branch-jobs/1.28.yaml](https://github.com/kubernetes/test-infra/pull/29137/files#diff-a8fa297d8acc9dc77022e002a57e9950a4ac8e2ea7e15689ca7d216d3b3ecea5) manually
3. Update other files accordingly based on 1.27 branch cut.

Meanwhile, we need to investigate why `make -C releng prepare-release-branch` create 1.28 without 1.28 branch cut.(Issue for tracking: https://github.com/kubernetes/test-infra/issues/29168)

File: releng/test_config.yaml [hunk 3]
```diff
@@ -252,29 +252,29 @@ k8sVersions:
     version: master
   beta:
     args:
-    - --extract=ci/latest-1.26
+    - --extract=ci/latest-1.28
```
Comment (cici37) [inline releng/test_config.yaml:255]: Should it be 1.27? (apply to following changes as well)

**Comment by ramrodo** [comment]
Closed this PR in favor of https://github.com/kubernetes/test-infra/pull/29172

**Pull request closed without merging**